tests/model/test_multivi.py

```diff
@@ -76,3 +85,416 @@ def test_multivi_single_batch():
     vae = MULTIVI(data, n_genes=50, n_regions=50)
     with pytest.warns(UserWarning):
         vae.train(3)
+
+
+def test_multivi_mudata_rna_prot_external():
+    # Example on how to download protein adata to mudata (from multivi tutotial) - mudata RNA/PROT
+    adata = scvi.data.pbmcs_10x_cite_seq()
+    adata.layers["counts"] = adata.X.copy()
+    sc.pp.normalize_total(adata)
+    sc.pp.log1p(adata)
+    adata.obs_names_make_unique()
+    protein_adata = ad.AnnData(adata.obsm["protein_expression"])
+    protein_adata.obs_names = adata.obs_names
+    del adata.obsm["protein_expression"]
+    mdata = MuData({"rna": adata, "protein": protein_adata})
+    sc.pp.highly_variable_genes(
+        mdata.mod["rna"],
+        n_top_genes=4000,
+        flavor="seurat_v3",
+        batch_key="batch",
+        layer="counts",
+    )
+    mdata.mod["rna_subset"] = mdata.mod["rna"][:, mdata.mod["rna"].var["highly_variable"]].copy()
+    mdata.update()
+    # mdata
+    # mdata.mod
+    MULTIVI.setup_mudata(
+        mdata,
+        rna_layer="counts",  # mean we use: mdata.mod["rna_subset"].layers["counts"]
+        protein_layer=None,  # mean we use: mdata.mod["protein"].X
+        batch_key="batch",  # the batch is here: mdata.mod["rna_subset"].obs["batch"]
+        modalities={
+            "rna_layer": "rna_subset",
+            "protein_layer": "protein",
+            "batch_key": "rna_subset",
```

Comment:
I'm confused about the batch_key. Does is need to be set up with modalities.

Reply:
the batch_key under modalities actually tells which modality stores batch_key in its obs data.
so for example:

```
    MULTIVI.setup_mudata(
        mdata,
        rna_layer="counts", 
        protein_layer=None, 
        batch_key="batch",  
        modalities={
            "rna_layer": "rna_subset",
            "protein_layer": "protein",
            "batch_key": "rna_subset",
        },
    )
```

the batch is here: mdata.mod["rna_subset"].obs["batch"]